### DISCUSSION TITLE
Conditional recording

### DISCUSSION THREAD

**Discussion by ArvyRogerio** [savonet/liquidsoap#2036] [Q&A]
I'm looking for a way to do a conditional recording.

I have a radio show called "Flash Hits" (dynamic playlist) and I want to use output.file to record it in MP3 format.

How can I do a conditional recording based on metadata? I know how to trigger based on source.on_metadata.

I don't know how to start and stop a source based on output.file(). Dynamic filename is not a problem.

Something like:

if (m['title']=="Flash Hits Intro") .... start output.file(%mp3(),...)
if (m['title']=="Flash Hits End") .... stop the previous source (source.shutdown()? how?)

Thanks!

**Comment by toots** [comment]
Hi!

This is a good question, thanks for asking. I was writing a response and realized some of the methods for it weren't included in the `2.0.0` release. This response applies to the `v2.0.1` bugfix branch, which should be out in a short while, maybe a couple of weeks.

With this release, you should be able to do something like this:

```ruby
# Source to record
s = ...

# Record the source, start disabled:
recorder = output.file(start=false, ...)

# Later in your script:
recorder.start()

# Then, when you want to stop:
recorder.stop()
```
